pkg/gatewayserver/io/udp/firewall_ratelimit_test.go

```diff
@@ -96,4 +96,28 @@ func TestRateLimitingFirewall(t *testing.T) {
 			})
 		}
 	})
+
+	t.Run("Recover", func(t *testing.T) {
+		a := assertions.New(t)
+		duration := 10 * test.Delay
```

Comment:
This is too short I think. You can add a bigger value. Ex: `(1 << 6)  * test.Delay`. You can check some other  tests in `pkg/gatewayserver`.